forge-game/src/main/java/forge/game/trigger/TriggerAbilityResolves.java

```diff
@@ -55,11 +55,12 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
     /** {@inheritDoc} */
     @Override
     public final void setTriggeringObjects(final SpellAbility sa, Map<AbilityKey, Object> runParams) {
-        final SpellAbility triggeredSA = (SpellAbility) runParams.get(AbilityKey.SpellAbility);
+        final SpellAbility triggeredSA = (SpellAbility) runParams.get(AbilityKey.CastSA);
         sa.setTriggeringObject(AbilityKey.Source, triggeredSA.getHostCard());
+        sa.setTriggeringObject(AbilityKey.SpellAbility, triggeredSA);
```

Comment:
maybe update the triggers to use CastSA instead of SpellAbility, so we don't need to use SpellAbility at all?

Reply:
Hmm, refactoring out SpellAbility here causes a bit of a tangle because a few cards are already wired up with `SVar:X:TriggeredSpellAbility$`...

If you think it would be worthwhile that might be a good issue for a future PR?

Reply:
I think I will do the opposite and just get rid of `AbilityKey.CastSA`
it's rarely used and doesn't really fit for all these trigger variants anyway

Reply:
hm CastSA -> SpellAbility is fine by me too i think